### PR TITLE
fix: Write version.json from the currently constructed package.json

### PR DESCRIPTION
**Pull request by Artur-**
The earlier logic was like
1. Read package.json
2. Modify package.json in memory
3. Generate version.json based on package.json file
4. Write modified package.json to file

Where step 3 ignored what was done in step 2

### CODE REVIEW

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdatePackagesNpmVersionLockingTest.java [hunk 1]
```diff
@@ -155,12 +155,44 @@ public void shouldDoNothingForNpmLocking_whenInPnpmMode()
...
+                classFinder.getResource(Constants.VAADIN_CORE_VERSIONS_JSON))
+                .thenReturn(null);
+
+        TaskUpdatePackages packageUpdater = createPackageUpdater(true);
+        JsonObject packageJson = packageUpdater.getPackageJson();
+        packageJson.getObject(DEPENDENCIES).put(TEST_DEPENDENCY,
+                PLATFORM_PINNED_DEPENDENCY_VERSION);
+        Assert.assertNull(packageJson.getObject(OVERRIDES));
+        FileUtils.write(packageUpdater.getPackageJsonFile(),
+                packageJson.toJson(), StandardCharsets.UTF_8);
```
Comment (caalador) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdatePackagesNpmVersionLockingTest.java:177]: We do not need to write the json file for this test anymore

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdatePackagesNpmVersionLockingTest.java [hunk 2]
```diff
@@ -155,12 +155,44 @@ public void shouldDoNothingForNpmLocking_whenInPnpmMode()
...
+
+        String versionsPath = packageUpdater.generateVersionsJson(packageJson);
+        File output = new File(packageUpdater.npmFolder, versionsPath);
+        Assert.assertTrue(
+                FileUtils.readFileToString(output, StandardCharsets.UTF_8)
+                        .contains(TEST_DEPENDENCY));
+
+        packageJson.getObject(DEPENDENCIES).remove(TEST_DEPENDENCY);
+        FileUtils.write(packageUpdater.getPackageJsonFile(),
+                packageJson.toJson(), StandardCharsets.UTF_8);
```
Comment (caalador) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdatePackagesNpmVersionLockingTest.java:187]: No need to write the package.json file for the test

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdatePackagesNpmVersionLockingTest.java (hunk 1 above)
Comment (Artur-) [reply flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdatePackagesNpmVersionLockingTest.java]: Removed

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdatePackagesNpmVersionLockingTest.java (hunk 2 above)
Comment (Artur-) [reply flow-server/src/test/java/com/vaadin/flow/server/frontend/NodeUpdatePackagesNpmVersionLockingTest.java]: Removed

**Review by caalador — APPROVED** [review]

**Pull request merged**